File: SERVER/src/main/java/com/youcode/petPlanet/entity/Product.java
package com.youcode.petPlanet.entity;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String description;
    private int quantity;
    private String image;
    private BigDecimal price;

    @ManyToOne
    private Category category;

    @OneToMany(mappedBy = "product")
    private List<PetProduct> petsProducts;

}

File: front/src/app/Interfaces/Product.ts
import { Category } from "./Category";
import { OrderLine } from "./OrderLine";
import { Pet } from "./Pet";

export interface Product {
    id: number;
    name: string;
    description: string;
    quantity: number;
    image: string;
    price: number;
    category: Category;
    pet: Pet;
  }